src/client.cpp

```diff
@@ -1360,8 +1360,22 @@ void Client::removeNode(v3s16 p)
 	}
 }
 
+/**
+ * Helper function for Client Side Modding
+ * Flavour is applied there, this should not be used for core engine
+ * @param p
+ * @param is_valid_position
+ * @return
+ */
 MapNode Client::getNode(v3s16 p, bool *is_valid_position)
 {
+	if (getCSMFlavourLimits() & CSMFlavourLimit::CSM_FL_LOOKUP_NODES) {
+		v3s16 ppos = floatToInt(m_env.getLocalPlayer()->getPosition(), BS);
+		if ((u32) std::abs(ppos.getDistanceFrom(p)) > m_csm_noderange_limit) {
```

Comment:
`getDistanceFrom` is never negative, unless you have imaginary content in the coordinates, which definitely isn't the case. abs and typecast are reduncant.